src/pages/auth.js

```diff
@@ -58,6 +59,52 @@ export class Auth extends React.Component {
     return {status: 307, redirectTo: '/'};
   }
 
+  constructor() {
+    super();
+    this.state = {
+      registerClicked: false,
+      headerHidden: false
+    };
+  }
+
+  componentDidMount() {
+    window.addEventListener('scroll', this.scrollHandler);
+  }
+
+  scrollHandler = () => {
+    if (this.state.registerClicked) return;
+
+  };
+
+  registerClickHandler = () => {
+    if (this.state.registerClicked) return;
+
+    this.setState({ registerClicked: true }); // TODO: description
+    this.cropLanding();
+  };
+
+  cropLanding() {
+    const landing = ReactDOM.findDOMNode(this.refs.landing);
+    const landingBody = ReactDOM.findDOMNode(this.refs.landingBody);
+    const header = document.querySelector('.landing__header-fixed');
+    const login = ReactDOM.findDOMNode(this.refs.login);
+    const heightBefore = landing.getBoundingClientRect().bottom;
+
+    landingBody.classList.add('landing__body-shortened');
+    landingBody.classList.remove('landing__body-fixed');
+    landing.classList.remove('layout__row-full');
+    header.classList.remove('landing__header-fixed');
+    header.classList.add('landing__header-abs');
+    landingBody.classList.add('landing__body-no_transition');
+    login.style.display = 'none';
```

Comment:
this line feels like not react'ish
you can achieve same effect by removing element from render.